IPython/html/static/widgets/js/widget.js

```diff
@@ -164,7 +200,7 @@ define(["widgets/js/manager",
         set_state: function (state) {
             var that = this;
             // Handle when a widget is updated via the python side.
-            return this._unpack_models(state).then(function(state) {
+            return Promise.resolve(state).then(function(state) {
```

Comment:
Isn't the resolve here redundant?  Above you use `return utils.resolve_promises_dict(state);` to resolve the state dict before a `then` which gets the resolved contents.